library/src/main/java/com/hippo/unifile/TreeDocumentFile.java

```diff
@@ -210,8 +210,7 @@ public UniFile[] listFiles(FilenameFilter filter) {
         final NamedUri[] uris = DocumentsContractApi21.listFilesNamed(mContext, mUri);
         final ArrayList<UniFile> results = new ArrayList<>();
         for (NamedUri uri : uris) {
-            String name = DocumentsContractApi19.getName(mContext, uri.uri);
-            if (name != null && filter.accept(this, name)) {
+            if (name != null && filter.accept(this, uri.name)) {
```

Comment:
Should it be this? Or should we still have the `getName` above? I tried checking out the branch locally and it failed to compile because `name` isn't declared.

```suggestion
            if (uri.name != null && filter.accept(this, uri.name)) {
```

Reply:
Oops! Yes, it should be `uri.name`.
Anyway, Imma close this in favor of #2, those optimization can be implemented there.

Reply:
Ok cool, I will do my best to merge in your changes to that branch.

Reply:
Oop, nevermind, since #2 got merged before I went back and updated things, it might make sense to re-open this PR so it can be merged separately.

Reply:
Opened #4 for that.